### ISSUE TITLE
[FunKey S][1.0.0.03] Games do not load despite the correct CRC32

### ISSUE THREAD

**Issue by DrUm78** [libretro/FBNeo#818]
Hardware: FunKey S (SoC V3s ARM Cortex A7-A clocked @ 1.2GHz)
Software: sdlretro for the frontent (https://github.com/FunKey-Project/sdlretro) and FBNeo 1.0.0.03 for the Libretro core

I can't launch any game, even if the CRCs match perfectly the ones in https://github.com/libretro/FBNeo/tree/master/dats. All it does is bringing me back to the desktop (GMenu2X). I compiled the core with the last sources ([955d9f7](https://github.com/libretro/FBNeo/commit/955d9f7c776c54fd082ce4459ffcde75d1ca3a65)), so 1.0.0.03 I guess.
Windows port works fine with 1.0.0.02 roms (this port has not been released in 1.0.0.03 yet so I can't compare with the exact same romset).

Here is the only error message I get in GMenu2X logs:
[07-05 08:36:26 I] Unhandled env: 0x22
[07-05 08:36:26 I] Unhandled env: 0x3E

Anyone has any idea about this issue?

Thanks.

**Comment by barbudreadmon** [comment]
This is not a core issue since the core works with dozens of platform, including arm SoCs. Your best bet is to ask support from the author(s) of that frontend.

**Comment by barbudreadmon** [comment]
I just learned that your device only has 64MB ram, if i had to guess i would say that's your problem here, try building a "light" core using those instructions : 
```
make -C src/burner/libretro LIGHT=1 generate-files
make -C src/burner/libretro LIGHT=1
```
It will remove support for tons of higher-end systems, making the core about 30% smaller. That's what's done for nintendo 3ds which has about the same memory limitation.

**Comment by DrUm78** [comment]
Thanks a lot for your help.
Unfortunately, even with the core shrinked from 48MB to 40MB, it still does not work and I got the same error message.
BTW, FBA 2012 works perfectly on my device, even for CPS3 games (not at 60fps though) and Libretro cores like ScummVM (70MB) works perfectly too.

**Comment by barbudreadmon** [comment]
Hmmm, i tried adding a new platform target, with the cflags i've seen at https://github.com/FunKey-Project/fbalpha2012-libretro/blob/master/svn-current/trunk/makefile.libretro, maybe it'll help : 
```
make -C src/burner/libretro platform=funkey generate-files
make -C src/burner/libretro platform=funkey
```

**Comment by DrUm78** [comment]
Nope it does not work. What you committed is in fact the same as the "classic_armv7_a7" which I already tried but thanks for your help on this, much appreciated!

**Comment by barbudreadmon** [comment]
>[07-05 08:36:26 I] Unhandled env: 0x22
[07-05 08:36:26 I] Unhandled env: 0x3E

If i had to guess, it means the frontend doesn't handle the 0x22 and 0x3E api calls, which would be subsystems and auto-frameskip. Those are optional and wouldn't be a problem on older RetroArch versions that don't support them, but maybe your frontend isn't handling missing api calls very well.

**Comment by DrUm78** [comment]
Ok, thanks for the info. Maybe the light Libretro frontend (SDLRetro) I use does not handle that well yes, thanks for the help!

**Comment by DrUm78** [comment]
@barbudreadmon For some reason, now the core launches fine on the FunKey S with the last commits. Anyway, it's way slower than FBA 2012 so I'll keep using this one, but just would like to let you know.

Thanks for your help.

**Comment by barbudreadmon** [comment]
Hmmm, i fixed an issue with macro code this morning, it was affecting neogeo games and fighting games with street fighter layouts, maybe that's the only games you were trying ? Apparently, that issue was causing random crashes on 32 bits android systems, maybe that funkey device is android-based ?

FWIW, there are tips that can help with performances : https://github.com/libretro/FBNeo/tree/master/src/burner/libretro#game-xxx-runs-slowly-why-

**Comment by DrUm78** [comment]
Yeah that one definitely fixed the issue: https://github.com/libretro/FBNeo/commit/a8a2a7cecc99c18061851d23acd59fe5a1ddd942
I also tried all the workaround to speed up emulation but as it's a more recent and accurate FBA emulation, it's way slower but that's not a problem for me, I will continue to use FBA 2012 ahnyway. At least now it works, thanks.

FYI I only tried X-Men: Children of the atom (CPS2) and the FunKey S is a Linux-based ARMv7 device (not Android though).
